File: Add Code Here/LeetCode_Sols/Easy/637. Average of Levels in Binary Tree/Solution.py
class Solution:
    def averageOfLevels(self, root: Optional[TreeNode]) -> List[float]:
        list = []
        q = [root]
        while len(q):
            n = len(q)
            sum = 0
            for i in range(n):
                node = q.pop(0)
                sum += node.val
                if node.left: q.append(node.left)
                if node.right: q.append(node.right)
            list.append(sum/n)
        return list

File: Add Code Here/LeetCode_Sols/Easy/637. Average of Levels in Binary Tree/Solution.cpp
//  Breadth First Search
class Solution {
public:
    vector<double> averageOfLevels(TreeNode* root) {
        vector<double> list;
        queue<TreeNode*> q;
        q.push(root);
        while(!q.empty()){
            int n = q.size();
            double sum = 0.0;
            for(int i = 0; i < n;  i++){
                TreeNode* node = q.front();
                q.pop();
                sum += node->val;
                if(node->left) q.push(node->left);
                if(node->right) q.push(node->right);
            }
            list.push_back(sum/n);
        }
        return list;
    }
};
